### DISCUSSION TITLE
Livewire runs really slowly....

### DISCUSSION THREAD

**Discussion by Peter-Fogarty** [livewire/livewire#4492] [1. Help] [answered]
Hi Team. Newbie to livewire, so please be gentle. I have a laravel 8 project, scaffolded with sanctum et al that all works. All that login and menu stuff uses livewire, so its good there. When I make my first attempt at wire:model binding, with the worlds simplest example.... 
```
<input wire:model="search" type="text" placeholder="Search..." />
    <div class="py-4">{{ $search ?? '' }}
```
and a component class of....
```
class Tradeboard extends Component
{
    public $search = "";

    public function mount()
    {
        Log::error("Hello from mount");
    }

    public function render()
    {

        return view('livewire.tradeboard');
    }
}
```
According to the XHR timings in both Chrome and Safari, it is taking almost a full 3 seconds from me pushing a key on the keyboard and then the livewire call returning to display the single letter on the screen. Everything is local, there is no database.... it feels like a timeout though.....  but I don know enough about livewire to know how to figure out why its so slow. By comparison the login functionality, which does use the database I have offsite in a private cloud, returns in less than a second..... 

Is is running on a new Apple Silicon Mac...... I even paid my money to sponsor this project hoping the video casts had some answers....

All help greatfully received.

**Comment by joshhanley** [comment]
@Peter-Fogarty hmm that's odd definitely shouldn't be getting those sorts of times!

I'm running on Apple Silicon using Laravel Valet and I get a response time of ~39ms for a text input, so I don't think it's a Livewire issue directly, it might be something to do with your setup.

What are you using to serve your app (valet/homestead/etc)?

![image](https://user-images.githubusercontent.com/882837/148719690-c93fabb2-42eb-496f-a5ba-058e7d0c8a4e.png)

**Reply by Peter-Fogarty** [reply]
I didn't presume it was livewire pre se, but was wondering what it was poking..... I am still technically a newbie with this stack.... so no mocking... but... I am just doing local testing through php artisan serve...... does that make me technically silly and no one actually does that?

**Reply by joshhanley** [reply]
@Peter-Fogarty nah that should also be quick! And plenty of people use that 🙂

You could try installing Laravel Debugbar and seeing if it has any further insights as to what might be happening?

**Reply by Peter-Fogarty** [reply]
I have it installed and it says nothing weird..... 
![Screen Shot 2022-01-10 at 4 41 30 pm](https://user-images.githubusercontent.com/53398296/148727259-accc91d8-b162-4233-a1c8-020b8008d448.png)

1 view, 3 queries, 1 model, 1 livewire.... just User auth objects.... and 2.98 seconds.... I am not sure where to look...
![Screen Shot 2022-01-10 at 4 46 04 pm](https://user-images.githubusercontent.com/53398296/148727282-fa057b04-8ce4-4082-9bf5-17fa9b366202.png)

Hence my cry for help, I was wondering if there was some install thing I did wrong that I didn't know I didn't know.......

**Reply by joshhanley** [reply]
@Peter-Fogarty yeah looks ok, although debugbar is also saying it's taking 2.64s, which suggests its something in your app and not your setup.

Can you screenshot your "Timeline" tab in debugbar? And also your full queries tab?

**Reply by Peter-Fogarty** [reply]
![Screen Shot 2022-01-11 at 10 19 50 am](https://user-images.githubusercontent.com/53398296/148859810-681a8c6c-fa16-483d-8c1d-d4d598189ed7.png)
![Screen Shot 2022-01-11 at 10 20 26 am](https://user-images.githubusercontent.com/53398296/148859840-8a94f72e-1847-46db-bad3-471b1124574d.png)

See now you say that and I look, it is clearly loosing heaps of time in the base laravel session handler..... I wonder what its doing..... and more to the point, i wonder if I made it do it somehow......

**Comment by amiranagram** [comment, marked as answer]
> ![Screen Shot 2022-01-11 at 10 19 50 am](https://user-images.githubusercontent.com/53398296/148859810-681a8c6c-fa16-483d-8c1d-d4d598189ed7.png) ![Screen Shot 2022-01-11 at 10 20 26 am](https://user-images.githubusercontent.com/53398296/148859840-8a94f72e-1847-46db-bad3-471b1124574d.png)
> 
> See now you say that and I look, it is clearly loosing heaps of time in the base laravel session handler..... I wonder what its doing..... and more to the point, i wonder if I made it do it somehow......

@Peter-Fogarty That's not a problem with Laravel or Livewire. It's your MySQL (assuming) server. The first query always takes the longest because of the time it needs to connect to the DB server. That will happen on every request because your DB connection is not persistent. But 1.95s is too long, so check your DB server, make sure it has enough memory to use, there are no errors on connection, etc.

**Reply by Peter-Fogarty** [reply]
Yeah, sorry, I should have posted back again... I moved to a local sqlite, and everything is better..... damn shared sql server on the other side of the world to me..... thanks @amiranagram and big thanks to @joshhanley .... I shall be sulking in the corner from now if you need me.  :-)

**Reply by DanielGSoftware** [reply]
Can I ask how you setup Mysql? Reading this got me wondering how it could take 1.95 sec to load

**Reply by joshhanley** [reply]
@Peter-Fogarty no worries 🙂 no need to sulk, it's how we learn 😁 at least now you know how to debug something like this!

@DanielGSoftware I think he didn't have a local mysql server, instead had a sql server hosted externally on a server on the other side of the world 😬 so a lot of that time would have been establishing the connection with the sql server.

**Reply by Peter-Fogarty** [reply]
Yes that's correct, I won't name and shame the provider, because you get what you pay for. I have been hosting email domains with them for 20 years, and they have a little bit of free MySql action, that has no promise of anything. I have been developing against that for years in different tech stacks, and was always use to it being slow, doing the best I could, and then waiting for production to make life wonderful. I guess the gotcha here, was that the database was being asked for my session details on every keypress, which for most people is not such a big deal. I just happened to have the cheapest DB hosting on the opposite side of the world to me. Not a problem for my emails... but a problem here.... it was part of my development disaster recovery plan have the DB miles away... now its on my laptop in a sqlite file... runs very fast....

**Reply by DanielGSoftware** [reply]
Alright makes sense, thank you and @joshhanley for responding!

**Reply by amiranagram** [reply]
@Peter-Fogarty You may want to consider installing a local MySQL server. SQLite has only limited features and sooner or later you will run into problems.
